pkg/apis/externaldns/types.go

```diff
@@ -162,7 +162,7 @@ var defaultConfig = &Config{
 	ExoscaleEndpoint:         "https://api.exoscale.ch/dns",
 	ExoscaleAPIKey:           "",
 	ExoscaleAPISecret:        "",
-	CRDSourceAPIVersion:      "externaldns.k8s.io/v1alpha",
+	CRDSourceAPIVersion:      "multiclusterdns.federation.k8s.io/v1alpha1",
```

Comment:
This is supposed to be generic. federation is just one use case. I am not sure whether this should be the default.

Reply:
Yeah, I agree this should be generic, but the CRDApiVersion here maybe should must set through the flag if the sourceis crd.